{{cookiecutter.project_name}}/pyproject.toml

```diff
@@ -1,139 +1,149 @@
+  "D104",
+  # __magic__ methods are often self-explanatory, allow missing docstrings
+  "D105",
+  # Missing docstring in __init__
+  "D107",
+  # <>: Disable one in each pair of mutually incompatible rules
+  # <> We don’t want a blank line before a class docstring
+  "D203",
+  # <> We want docstrings to start immediately after the opening triple quote
+  "D213",
+  # first line should end with a period [Bug: doesn’t work with single-line docstrings]
+  "D400",
+  # First line should be in imperative mood; try rephrasing
+  "D401",
+  # line too long -> we accept long comment lines; formatter gets rid of long code lines
+  "E501",
+  # Do not assign a lambda expression, use a def -> lambda expression assignments are convenient
+  "E731",
+  # allow I, O, l as variable names -> I is the identity matrix
+  "E741",
+]
+lint.per-file-ignores."*/__init__.py" = [
+  "F401",
+]
+lint.per-file-ignores."docs/*" = [
+  "I",
+]
+lint.per-file-ignores."tests/*" = [
+  "D",
 ]
+lint.pydocstyle.convention = "numpy"
 
 [tool.pytest.ini_options]
-testpaths = ["tests"]
+testpaths = [ "tests" ]
 xfail_strict = true
 addopts = [
-    "--import-mode=importlib",  # allow using test files with same name
+  "--import-mode=importlib", # allow using test files with same name
 ]
 
-[tool.ruff]
-line-length = 120
-src = ["src"]
-extend-include = ["*.ipynb"]
-
-[tool.ruff.format]
-docstring-code-format = true
-
-[tool.ruff.lint]
-select = [
-    "F",  # Errors detected by Pyflakes
-    "E",  # Error detected by Pycodestyle
-    "W",  # Warning detected by Pycodestyle
-    "I",  # isort
-    "D",  # pydocstyle
-    "B",  # flake8-bugbear
-    "TID",  # flake8-tidy-imports
-    "C4",  # flake8-comprehensions
-    "BLE",  # flake8-blind-except
-    "UP",  # pyupgrade
-    "RUF100",  # Report unused noqa directives
-]
-ignore = [
-    # line too long -> we accept long comment lines; formatter gets rid of long code lines
-    "E501",
-    # Do not assign a lambda expression, use a def -> lambda expression assignments are convenient
-    "E731",
-    # allow I, O, l as variable names -> I is the identity matrix
-    "E741",
-    # Missing docstring in public package
-    "D104",
-    # Missing docstring in public module
-    "D100",
-    # Missing docstring in __init__
-    "D107",
-    # Errors from function calls in argument defaults. These are fine when the result is immutable.
-    "B008",
-    # __magic__ methods are often self-explanatory, allow missing docstrings
-    "D105",
-    # first line should end with a period [Bug: doesn't work with single-line docstrings]
-    "D400",
-    # First line should be in imperative mood; try rephrasing
-    "D401",
-    ## Disable one in each pair of mutually incompatible rules
-    # We don’t want a blank line before a class docstring
-    "D203",
-    # We want docstrings to start immediately after the opening triple quote
-    "D213",
+[tool.coverage.run]
+source = [ "{{ cookiecutter.package_name }}" ]
+omit = [
```

Comment:
Are all these rearrangements necessary? This will create one huge merge conflict when making the template-update PRs

Reply:
That’s the thing: pyproject-fmt dictates the order of these blocks.

If it was possible to run the hooks before creating the diff, maybe we could avoid conflicts